include/matx/core/operator_utils.h

```diff
@@ -120,7 +120,7 @@ namespace matx {
     __MATX_HOST__ __MATX_INLINE__ void AllocateTempTensor(TensorType &tensor, Executor &&ex, ShapeType &&shape, typename TensorType::value_type **ptr) {
       const auto ttl_size = std::accumulate(shape.begin(), shape.end(), static_cast<index_t>(1),
                                   std::multiplies<index_t>()) * sizeof(typename TensorType::value_type);      
-      if constexpr (is_cuda_executor_v<Executor>) {
+      if constexpr (is_cuda_executor_v<Executor> || is_stf_executor_v<Executor>) {
```

Comment:
Should we make a separate `is_device_executor`?

Reply:
Perhaps both stf and cuda executor needs to inherit from something common ? A trait class like is_device_executor is a good strategy to minimize the changes in the code to start with ...